src/ducks/SANCharts/ChartPage.js

```diff
@@ -45,31 +53,35 @@ const LoadableChartMetricsTool = Loadable({
 })
 
 const getChartInitialState = props => {
-  let passedState
-  if (props.location && props.location.search) {
-    const data = qs.parse(props.location.search, { arrayFormat: 'comma' })
-    if (typeof data.metrics === 'string') {
-      data.metrics = [data.metrics]
-    }
-    passedState = data
-  } else {
-    let { slug, from, to, title, timeRange, metrics, interval } = props
-
-    if (!from) {
-      const { from: f, to: t } = getIntervalByTimeRange(timeRange)
-      from = f.toISOString()
-      to = t.toISOString()
-      interval = getNewInterval(from, to)
-    }
-    passedState = {
-      slug,
-      title,
-      metrics,
-      from,
-      to,
-      timeRange,
-      interval
+  let { slug, from, to, title, timeRange, metrics, interval } = props
+
+  if (!from) {
+    const { from: f, to: t } = getIntervalByTimeRange(timeRange)
+    from = f.toISOString()
+    to = t.toISOString()
+    interval = getNewInterval(from, to)
+  }
+  let passedState = {
+    slug,
+    title,
+    metrics,
+    from,
+    to,
+    timeRange,
+    interval
+  }
+
+  if (props.location) {
+    const parsedQuery = mapParsedTrueFalseFields(
+      qs.parse(props.location.search, {
+        arrayFormat: 'comma'
+      })
+    )
+    if (typeof parsedQuery.metrics === 'string') {
+      parsedQuery.metrics = [parsedQuery.metrics]
     }
+
+    passedState = { ...passedState, ...parsedQuery }
```

Comment:
Either initial state is passed as a `props` or as a `search query`. Mixing them results code vulnerable to bugs. Return to the previous code structure, where these cases are handled separately

Reply:
Will change. But a lot of 'if' structures make code more unreadable, if it is not look as pipe(read props and then apply data from query).